src/main/java/core/mate/academy/service/BulldozerProducer.java

```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer<T> implements MachineProducer<T> {
```

Comment:
BulldozerProducer is exact implementation of MachineProducer. You should use concrete type instead of generic. Fix it for all producers 
```suggestion
public class BulldozerProducer implements MachineProducer<Bulldozer> {
```